site/profile/manifests/gpu.pp

```diff
@@ -84,6 +92,48 @@
       seltype => 'lib_t'
     }
   }
+
+  # WARNING : since the fact is computed before Puppet agent run,
+  # on a clean host, the  symbolic links to the NVIDIA libraries
+  # that include the version number will be created on the
+  # second Puppet run only.
+  $driver_vers = $::facts['nvidia_driver_version']
+  if $driver_vers != '' {
+    $nvidia_libs_vers = [
+      "libcuda.so.${driver_vers}",
+      "libEGL_nvidia.so.${driver_vers}",
+      "libGLESv1_CM_nvidia.so.${driver_vers}",
+      "libGLESv2_nvidia.so.${driver_vers}",
+      "libGLX_nvidia.so.${driver_vers}",
+      "libnvcuvid.so.${driver_vers}",
+      "libnvidia-cbl.so.${driver_vers}",
+      "libnvidia-cfg.so.${driver_vers}",
+      "libnvidia-compiler.so.${driver_vers}",
+      "libnvidia-eglcore.so.${driver_vers}",
+      "libnvidia-encode.so.${driver_vers}",
+      "libnvidia-fatbinaryloader.so.${driver_vers}",
+      "libnvidia-fbc.so.${driver_vers}",
+      "libnvidia-glcore.so.${driver_vers}",
+      "libnvidia-glsi.so.${driver_vers}",
+      "libnvidia-glvkspirv.so.${driver_vers}",
+      "libnvidia-ifr.so.${driver_vers}",
+      "libnvidia-ml.so.${driver_vers}",
+      "libnvidia-opencl.so.${driver_vers}",
+      "libnvidia-opticalflow.so.${driver_vers}",
+      "libnvidia-ptxjitcompiler.so.${driver_vers}",
+      "libnvidia-rtcore.so.${driver_vers}",
+      "libnvidia-tls.so.${driver_vers}",
+      "libnvoptix.so.${driver_vers}"
+    ]
+
+    $nvidia_libs_vers.each |String $lib| {
+      file { "/usr/lib64/nvidia/${lib}":
```

Comment:
Can we make the location `/usr/lib64/nvidia` configurable? EESSI has decided to go with `/opt/eessi/lib` to look for additional libraries

Reply:
Sure. Note that Compute Canada is reviewing the need for that folder given that the file `fatbinaryloader.so.$version` no longer exist since it is now part of the NVIDIA driver instead.

Some testing is required, but the process is getting documented here:
https://github.com/ComputeCanada/software-stack/issues/58

Reply:
The path is now configurable and is configured to `/opt/eessi/lib` in `software_stack/eessi.yaml`.

Reply:
Excellent, thanks a lot